experiments/conductor/cmd/runner/utilities.go

```diff
@@ -801,7 +825,7 @@ func processBranch(ctx context.Context, opts *RunnerOptions, branch Branch, skip
 	close := setLoggingWriter(opts, branch)
 	defer close()
 
-	checkoutBranch(ctx, branch, opts.branchRepoDir)
+	checkoutBranchWithoutuncommittedChanges(ctx, branch, opts.branchRepoDir)
```

Comment:
Can we separate this into 2 functions? Having a clean working directory or abort function seems useful beyond this use case.

Reply:
Done.